docs/TOPOLOGY.md

```diff
@@ -0,0 +1,45 @@
+# Process Topology
+```
+Note: this document describes the connections between applications, drivers, datastores, and nodes. It doesn't go into the details of the p2p connections between nodes.
+```
+
+One of the core functions of each community's state machine is to keep track of all the processes running on that community. There are several main types of processes managed by Habitat's process manager:
+* Applications
+* Datastore drivers
+* Datastores
+
+In addition, the topology must track which nodes these different processes are running on.
+
+To keep the network topology of interconnected processes in Habitat easy to understand, a set of simple rules are enforced by all Habitat nodes. These rules define which processes are allowed to comunicate with which other ones.
+
+## Rules
+The following is the basic set of rules that apply to the different types of vertexes in the topology:
+* Nodes
+* Processes(applications, datastores, drivers)
+* Process Instances (...)
+
+### Process rules:
+* Processes cannot be connected to nodes
+* Each process must have at least one instance of the same type
```

Comment:
why does this need to be true? can a process be "not running"? or maybe this is encapsulated in the process instance?